### PR TITLE
10msbuffer

### PR DESCRIPTION
**Pull request by Vamshigopal**

### CODE REVIEW

**Comment by sathyap-chrome** [comment]
We are working on making 10ms default for speaker playback (assuming 3p supports). We are starting with designs not productized. 

We see power improvement with this change.

PS : We pushed the commit on top of main's last commit - not sure why there is a conflict. Will look in to this again.

**Comment by marc-hb** [comment]
@sathyap-chrome we've seen Github reporting spurious git conflicts before. I don't have a solution unfortunately.

**Comment by plbossart** [comment]
@sathyap-chrome @sathya-nujella 10ms for each period is a very large increase to the latency for playback. Last time I tried basic SSP playback, I measured a 4ms delay, that would be 40ms delay now. That's quite large.
Is this even remotely compatible with other requirements such as round-trip latency and lip-sync?

**Comment by marc-hb** [comment]
The git conflicts are real. As of 6554ecdf206151fc325ed2d12a63ee2c6f802b82, this PR has clear conflicts with 1e3a4cfe9b26 which is already merged.

**Comment by sathya-nujella** [comment]
> @sathyap-chrome @sathya-nujella 10ms for each period is a very large increase to the latency for playback. Last time I tried basic SSP playback, I measured a 4ms delay, that would be 40ms delay now. That's quite large. Is this even remotely compatible with other requirements such as round-trip latency and lip-sync?

Hi @plbossart : 
1) is this delay of "40ms for each 10ms period" coming from host-to-firmware DMA's or within firmware?
2) We had few ADL based models productized with 10ms period, we did not hear AV sync issues. Thank you!

@cujomalainey - please share if any reports on AV synchronization/lip-sync issues with period size changes to 10ms.

**Comment by plbossart** [comment]
@sathya-nujella I was talking about the firmware only. If you use 2 periods per component and multiple components very quickly the latency will add up.

I am very very surprised that this was accepted for ADL, and that can't possibly be the default solution moving forward. If we want to reduce power, the right thing to do is to increase the latency on a dedicated PCM device (i.e. Deep-buffer), but keep the low-latency capability. Audio is all about latency, increasing buffer sizes will hit you back big time.

**Comment by sathyap-chrome** [comment]
@marc-hb  will look in to the conflict. thanks.

File: tools/topology/topology1/sof-adl-nau8825.m4 [hunk 1]
```diff
@@ -93,6 +93,8 @@ ifdef(`BT_OFFLOAD', `
 # PCM99 <---- volume <---- DMIC01 (dmic 48k capture)
 # PCM100 <---- kpb <---- DMIC16K (dmic 16k capture)
 
+ifdef(`SPK_MIC_PERIOD_US', `define(`SPK_MIC_PERIOD_US', 10000)', `define(`SPK_MIC_PERIOD_US', 1000)')
```
[the remark this answers is not in the document]
Comment (sathyap-chrome) [reply tools/topology/topology1/sof-adl-nau8825.m4:96]: Agreed, will do that

[the remark this answers is not in the document]
Comment (sathyap-chrome) [reply tools/topology/topology1/sof-adl-nau8825.m4]: @lgirdwood  Done.

**Comment by cujomalainey** [comment]
Hi @plbossart 

Agreed, the idea is not great at face value, but there is logic here. We intend to deploy our AEC which needs a 10ms period. When we deploy our AEC on SOF vs CRAS, we actually see a gain reduction because it can keep a much narrower window when looking for echos. Note this only applies to the capture path. We didn't get the chance to test out 1ms on playback and have it feed into a 10ms buffer on the demux. For playback the plan is a bit more complex so I would be happy to explain it over VC.

**Comment by plbossart** [comment]
Well, you've got this big issue that we can only support one system tick @cujomalainey. If you start having different periods for different processing, e.g. 10ms on capture and 1ms on playback, the processing will have to complete before the 1ms playback DAI tick expires. We don't have good support for processing that requires buffering for frame-based processing - that's why the deep-buffer is a problem already.

And even if you save on the echo cancellation search window the latency problem remains. What you gain on one side is offset by a loss on the other.

I understand where you are coming from but the suggested 10ms period increase is a sledgehammer that will break many eggs.

**Comment by cujomalainey** [comment]
@plbossart understood and I actually remembered why we did 10ms playback which is exactly that tick issue. We won't be able to get around it till we have mult-core pipelines.

And agreed on the offset, our team sees the value as something we are willing to pay currently to get the better AEC performance. 

Agreed, it would be better if we were in a state that was already supporint low latency and deep buffer playback, but short 10ms pipelines appear to be a good middle ground given the currently latency introduced by the rest of the stack.

@chiang831 FYI

**Comment by plbossart** [comment]
Sounds good @cujomalainey, if everyone is aware of limitations and next steps/improvements that's fine with me. 

I was concerned with the commit message that hinted at a sustained direction to keep 10ms periods, and clearly that's just a temporary work-around for specific products.

**Comment by cujomalainey** [comment]
@perahgren also FYI

Yes this is not planned for zephyr based devices on PB. Capture we will have to play with multicore to get the balance to an optimal level

**Comment by sathyap-chrome** [comment]
Thanks @plbossart and @cujomalainey for the detailed discussion.
Can we get the PR now as a solution for non zephyr based devices ( ADL and RPL)

**Comment by sathyap-chrome** [comment]
@lgirdwood / @cujomalainey / @plbossart i've rebased to latest. Since we are aligned on this change. Can you approve the same.

**Review by cujomalainey — APPROVED** [review]

**Pull request merged**